src/Microsoft.Health.Fhir.Core/Features/Search/SearchIndexEntryExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.ValueSets;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search
+{
+    internal static class SearchIndexEntryExtensions
+    {
+        public static void SetMinMaxValues(this IEnumerable<SearchIndexEntry> entries)
+        {
+            // determine min/max values
+            foreach (var item in entries.GroupBy(x => x.SearchParameter.Name))
+            {
+                switch (item.First().SearchParameter.Type)
+                {
+                    case SearchParamType.String:
```

Comment:
I think we should only handle type we know we can/should support. For example, what does it mean to sort by quantity when there are different units, systems, etc. 
I would do string  and date for now